ReferralPlease/ReferralPlease/ViewController.swift

```diff
@@ -141,93 +158,136 @@ extension ViewController: WKNavigationDelegate {
-            let tokenURLFull = "https://api.linkedin.com/v2/me?projection=(id,firstName,lastName,profilePicture(displayImage~:playableStreams))&oauth2_access_token=\(accessToken)".addingPercentEncoding(withAllowedCharacters: .urlQueryAllowed)
-            let verify: NSURL = NSURL(string: tokenURLFull!)!
+
+            let tokenURLFull = "https://api.linkedin.com/v2/me?projection=(id,firstName,lastName,profilePicture,emailAddress(displayImage~:playableStreams))&oauth2_access_token=\(accessToken)".addingPercentEncoding(withAllowedCharacters: .urlQueryAllowed)
+                guard let verify: NSURL = NSURL(string: tokenURLFull ?? "") else {
+                    print("Invalid URL")
+                    return}
+   
             let request: NSMutableURLRequest = NSMutableURLRequest(url: verify as URL)
             let task = URLSession.shared.dataTask(with: request as URLRequest) { data, response, error in
+
                 if error == nil {
-                    let linkedInProfileModel = try? JSONDecoder().decode(LinkedInProfileModel.self, from: data!)
+                    guard let jsonData = data else {
+                        print("No data found")
+                        return
+                    }
                     
-                    //AccessToken
-                    print("LinkedIn Access Token: \(accessToken)")
+
+
                     
-                    // LinkedIn Id
-                    let linkedinId: String! = linkedInProfileModel?.id
-                    print("LinkedIn Id: \(linkedinId ?? "")")
-
-                    // LinkedIn First Name
-                    let linkedinFirstName: String! = linkedInProfileModel?.firstName.localized.enUS
-                    print("LinkedIn First Name: \(linkedinFirstName ?? "")")
-
-                    // LinkedIn Last Name
-                    let linkedinLastName: String! = linkedInProfileModel?.lastName.localized.enUS
-                    print("LinkedIn Last Name: \(linkedinLastName ?? "")")
-
-                    // LinkedIn Profile Picture URL
-                    let linkedinProfilePic: String!
-
-                    /*
-                     Change row of the 'elements' array to get diffrent size of the profile url
-                     elements[0] = 100x100
-                     elements[1] = 200x200
-                     elements[2] = 400x400
-                     elements[3] = 800x800
-                    */
-                    if let pictureUrls = linkedInProfileModel?.profilePicture.displayImage.elements[2].identifiers[0].identifier {
-                        linkedinProfilePic = pictureUrls
-                    } else {
-                        linkedinProfilePic = "Not exists"
+                    do {
+            
+    
+                        let linkedInProfileModel = try JSONDecoder().decode(LinkedInProfileModel.self, from: jsonData)
+                        self.db?.collection("users").document(linkedInProfileModel.id).setData([
```

Comment:
@justinlim1 if I login the second time, does this overwrite the old data or will it add a duplicate?

Reply:
Its indexed by linkedin id so the document will just update unless linkedin id changes

Reply:
nice